src/components/TabComponent/TabComponent.js

```diff
@@ -0,0 +1,33 @@
+import React, {  PureComponent } from "react";
+import "./TabComponent.css";
+
+export default class TabComponent extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { activeTab: 0 };
+    //    this.setState(this.state);
+    this.onTabChange = (index) => {
+      this.setState({ activeTab: index });
+      this.props.tabChanged(this.state.activeTab);
+    };
+  }
+
+  render() {
+    return (
+      <div className="tab-container">
+        {this.props.tabs.map((value, index) => {
+          return (
+            <div key={index}
+              onClick={()=>this.onTabChange(index)}
+              className={
```

Comment:
if you want to apply more than 1 class and dinamicaly calculate className better to use package classnames
https://www.npmjs.com/package/classnames

and usage this package:

```
import classnames from 'classnames';
........

render() {
const { tabs } = this.props;
  const tabClassNames = classnames('tab', {
    activeTab: index === this.state.activeTab,
  });

  {tabs.map((value, index) => {
     return (
       <div
         key={index}
         onClick={()=>this.onTabChange(index)}
         className={tabClassNames}
      >
        {value}
     </div>
.........
}
```

Reply:
here i am not able to pass the index to the classNames function